src/Infrastructure/Validator/Manifest/TaskManifestConfiguration.php

```diff
@@ -0,0 +1,309 @@
+                    ->end()
+                ->end()
+                ->scalarNode('command')
+                    ->isRequired()
+                    ->validate()
+                        ->ifEmpty()
+                        ->thenInvalid('Task command is require.')
+                    ->end()
+                ->end()
+                ->scalarNode('help')
+                    ->defaultNull()
+                ->end()
+                ->scalarNode('type')
+                    ->isRequired()
+                    ->validate()
+                        ->ifNotInArray([Task::TASK_TYPE_LOCAL_CLI, Task::TASK_TYPE_LOCAL_CLI_INTERACTIVE])
+                        ->thenInvalid(
+                            vsprintf(
+                                'Task should have %s or %s.',
+                                [Task::TASK_TYPE_LOCAL_CLI, Task::TASK_TYPE_LOCAL_CLI_INTERACTIVE],
+                            ),
+                        )
+                    ->end()
+                ->end()
+                ->arrayNode('report_converter')
+                    ->children()
+                        ->scalarNode('name')
+                            ->isRequired()
+                            ->validate()
+                                ->ifTrue(
+                                    function ($name) {
+                                        return !$this->validationHelper->isConverterValid($name);
+                                    },
+                                )
+                                ->thenInvalid('Converter name `%s` doesn\'t exist.')
+                            ->end()
+                        ->end()
+                        ->arrayNode('configuration')
+                            ->children()
+                                ->scalarNode('input_file')
+                                    ->isRequired()
+                                ->end()
+                                ->scalarNode('producer')
+                                    ->isRequired()
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+            ->validate()
+                ->ifTrue(function (array $task) {
+                    return !$this->validationHelper
+                        ->isPlaceholderInStringValid(
+                            $task['command'],
+                            array_map(function (array $placeholder) {
+                                return $placeholder['name'];
+                            }, $task['placeholders']),
+                        );
+                })
+                ->thenInvalid('Not all placeholders uses.')
+            ->end();
+
+        $this->addPlaceholderDefinition(
+            $node
+            ->children()
+                ->arrayNode('placeholders')
+                ->arrayPrototype()
+                    ->children(),
+        );
+        $lifecycle = $node
```

Comment:
Please, extract the lifecycle tree builder to the separate method as you did for the placeholders. The same for the task set configuration.